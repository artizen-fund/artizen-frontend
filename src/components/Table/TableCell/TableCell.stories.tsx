import TableCell from './'

const story = {
  title: 'tables/TableCell',
  component: TableCell,
  argTypes: {},
}
export default story

export const TableCellComponent = (props: any) => <TableCell {...props}>derp</TableCell>
