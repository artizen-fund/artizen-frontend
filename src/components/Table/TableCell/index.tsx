import styled from 'styled-components'
import { breakpoint, palette } from '@theme'
import { rgba } from '@lib'

export interface TableCellProps {
  children: React.ReactNode
}

const TableCell = styled.div<TableCellProps>`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 40px;
  padding: 4px 12px 4px 12px;
  @media only screen and (min-width: ${breakpoint.laptop}px) {
    height: 48px;
    padding: 4px 16px 4px 16px;
  }
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    height: 56px;
    padding: 4px 24px 4px 24px;
  }

  border-radius: 8px;
  background-color: ${rgba(palette.stone, 0.24)};
  @media (prefers-color-scheme: dark) {
    background-color: ${rgba(palette.barracuda, 0.24)};
  }
`

export default TableCell
