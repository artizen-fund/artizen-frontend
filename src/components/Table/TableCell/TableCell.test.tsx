import { render } from '@testing-library/react'
import TableCell from './'

it('renders TableCell unchanged', () => {
  const { container } = render(<TableCell>derp</TableCell>)
  expect(container).toMatchSnapshot()
})
