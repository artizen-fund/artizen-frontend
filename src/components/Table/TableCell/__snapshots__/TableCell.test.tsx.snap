// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TableCell unchanged 1`] = `
<div>
  <div
    class="sc-c0aed08d-0 dWVRiz"
  >
    derp
  </div>
</div>
`;
